src/Elasticsearch.Net/Connection/MetaData/MetaDataHeaders.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Licensed to Elasticsearch B.V under one or more agreements.
+// Elasticsearch B.V licenses this file to you under the Apache 2.0 License.
+// See the LICENSE file in the project root for more information
+
+namespace Elasticsearch.Net
+{
+	internal sealed class MetaDataHeaders : IMetaDataHeaders
+	{
+		public MetaDataHeaders(ClientVersionInfo clientVersionInfo)
+		{
+			AsyncMetaDataHeader = new MetaDataHeader(clientVersionInfo, "es", true);
+			SyncMetaDataHeader = new MetaDataHeader(clientVersionInfo, "es", false);
+		}
+
+		public MetaDataHeader AsyncMetaDataHeader { get; private set; }
```

Comment:
```suggestion
		public MetaDataHeader AsyncMetaDataHeader { get; }
```

Reply:
Maybe a good first `record` ? 😄

Reply:
Yeah it could be. I want to read up on using records for older targets since I believe we may need to include code for a couple of attributes which may be required for them to work after compilation.